eureka2-integration/src/test/java/com/netflix/eureka2/integration/startup/FileBasedServerStartupAndShutdownIntegrationTest.java

```diff
@@ -0,0 +1,23 @@
+package com.netflix.eureka2.integration.startup;
+
+import com.netflix.eureka2.server.EurekaWriteServer;
+import org.junit.Test;
+
+/**
+ * Since this test relies on Archaius which has some global settings, we can't have different tests for both read
+ * and write servers under the same JVM. As the tests are testing startup and shutdown behaviour that is common to
+ * both (within AbstractEurekaServer), we just pick one of the two server types (write servers) to test this.
+ *
+ * @author David Liu
+ */
+public class FileBasedServerStartupAndShutdownIntegrationTest extends WriteServerStartupAndShutdownIntegrationTest {
```

Comment:
It's in a separate class for clearer organisation. If we added it to the write test class chances are someone in the future might see the "missing" complementary test in the read class and add a read version back, which will then get us back to the same problem as now.